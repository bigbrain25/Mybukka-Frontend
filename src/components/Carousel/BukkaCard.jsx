import React, { Fragment } from 'react';

import PropTypes from 'prop-types';

import generateSmImageUrl from 'Utilities/generateSmallScreenImageUrl';
import MarkIcon from 'Icons/Remark';
import Navlink from 'Components/navlink/Navlink';
import Price from 'Components/badge/Price';

import './BukkaCard.scss';

const TopOverlayText = ({ heading, subHeading }) => (
  <div className="overlay-top">
    <h4>{heading}</h4>
    <h5>{subHeading}</h5>
  </div>
);

const BottomOverlayText = ({ heading, subHeading }) => (
  <div className="overlay-bottom">
    <h4 className={
      `${subHeading ? 'overlay-h4' : 'overlay-h4s'}`}
    >{heading}</h4>
    {subHeading && <h5>{subHeading}</h5>}
  </div>
);

const TextOverlay = ({ top, bottom, heading, subHeading }) => (
  <Fragment>
    {top && <TopOverlayText heading={heading} subHeading={subHeading} />}
    {bottom && <BottomOverlayText heading={heading} subHeading={subHeading} />}
  </Fragment>
);

const NormalText = ({ deliveryCost, deliveryTime, rating }) => (
  <div className="details">
    <h3>
      <span className="details-location">
        San francisco
        <span>
          kitchen
          <span className="icon">
            <MarkIcon />
          </span>
        </span>
      </span>
      {rating && <span className="rating">{rating}</span>}
    </h3>
    <div className="delivery">
      {deliveryCost ? <Price price={deliveryCost} /> : null}
      {deliveryCost && deliveryTime ? <i className="dot-spacing">.</i> : null}
      <div className="delivery-time-text">{deliveryTime}</div>
    </div>
  </div>
);

const BukkaCard = ({
  imageUrl,
  deliveryCost,
  deliveryTime,
  rating,
  imageHeight,
  textOverlay,
  top,
  bottom,
  heading,
  subHeading
}) => {
  const smImgUrl = generateSmImageUrl(imageUrl);

  return (
    <div className="mt-4 bukka-card">
      <img
        className={`img-small-screen bukka-img ${imageHeight}`}
        src={smImgUrl}
        alt="alt_image"
      />
      <img
        className={`img-large-screen bukka-img ${imageHeight}`}
        src={imageUrl}
        alt="alt_image"
      />
      {textOverlay && (
        <TextOverlay
          top={top}
          bottom={bottom}
          heading={heading}
          subHeading={subHeading}
        />
      )}
      {!textOverlay && (
        <NormalText
          deliveryCost={deliveryCost}
          deliveryTime={deliveryTime}
          rating={rating}
        />
      )}
    </div>
  );
};

const GetBukka = ({ classNames, href, ...props }) => (
  <div className={`card-container ${classNames}`}>
    <div className="card-wrap">
      <Navlink classNames="link" href={href}>
        <BukkaCard {...props} />
      </Navlink>
    </div>
  </div>
);

export default GetBukka;

NormalText.defaultProps = {
  deliveryTime: '',
  rating: '',
  deliveryCost: 0
};

NormalText.propTypes = {
  deliveryCost: PropTypes.number,
  deliveryTime: PropTypes.string,
  rating: PropTypes.string
};

TextOverlay.propTypes = {
  top: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]).isRequired,
  bottom: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]).isRequired,
  heading: PropTypes.string.isRequired,
  subHeading: PropTypes.string.isRequired
};

TopOverlayText.propTypes = {
  heading: PropTypes.string.isRequired,
  subHeading: PropTypes.string.isRequired
};

BottomOverlayText.defaultProps = {
  subHeading: ''
};

BottomOverlayText.propTypes = {
  heading: PropTypes.string.isRequired,
  subHeading: PropTypes.string,
};

BukkaCard.defaultProps = {
  deliveryTime: '',
  rating: '',
  imageHeight: '',
  textOverlay: false,
  deliveryCost: 0,
  top: false,
  bottom: false,
  heading: 'Free Delivery',
  subHeading: ''
};

BukkaCard.propTypes = {
  imageUrl: PropTypes.string.isRequired,
  deliveryCost: PropTypes.number,
  deliveryTime: PropTypes.string,
  rating: PropTypes.string,
  imageHeight: PropTypes.string,
  textOverlay: PropTypes.bool,
  top: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
  bottom: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
  heading: PropTypes.string,
  subHeading: PropTypes.string
};

GetBukka.defaultProps = {
  classNames: '',
  href: '/feed'
};

GetBukka.propTypes = {
  classNames: PropTypes.string,
  href: PropTypes.string
};
