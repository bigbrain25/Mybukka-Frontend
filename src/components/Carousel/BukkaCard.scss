@import '../../stylebase/variables.scss';
$fontFamily: 'Roboto', sans-serif;

.card-container {
  position: relative;
  padding-bottom: 20px;
}

.default-overlay {
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  top: 10px;
  right: 12px;
  left: 0;
}

.bukka-img-color {
  background: rgb(233, 231, 231);
}

.card-wrap {
  display: flex;
  flex-shrink: 0;
  flex-direction: column;
  cursor: pointer;
  position: relative;
}

.bukka-card {
  position: relative;
  background-color: transparent;
  display: flex;
  overflow: hidden;
}

.touchdown {
  .rating {
    display: none;
  }
}

.details {
  position: relative;
  padding-top: 16px;
}

.details h3 {
  display: flex;
  margin-top: 0px;
  margin-bottom: 4px;
  font-family: $fontFamily;
  font-weight: 500;
  font-size: 16px;
  text-transform: capitalize;
  @media (max-width: 767px) {
    font-size: 13px;
  }
}

.collection-image-spacing {
  height: auto;
  padding-bottom: 58.82352941176471%;
}

.normal-image-spacing {
  height: auto;
  padding-bottom: 61.1%;
}

.cuisine-image-spacing {
  height: auto;
  padding-bottom: 119.04761904761905%;
}

.category-image-spacing {
  height: auto;
  padding-bottom: 105.047619%;
}

.custom-bukka-image {
  position: absolute;
  background-position: center center;
  background-size: cover;
  background-repeat: no-repeat;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  -webkit-transition: opacity 0.4s linear;
  transition: opacity 0.4s linear;
  background-color: rgb(233, 231, 231);
}


.custom-delivery-badge {
  color: #9c1c26;
  font-size: 14px;
  font-weight: 500;
  letter-spacing: 0.72px;
  position: relative;
  top: -4px;
  padding-left: 5px;
}

.delivery {
  display: flex;
  flex-direction: row;
  font-family: $fontFamily;
  font-size: 14px;
  letter-spacing: 0.14px;
  font-weight: 400;
  line-height: 24px;
  margin-top: 10px;
  color: #8f95a3;
}

.bukka-card:hover .favorite {
  opacity: 1;
}

.other-bukka-card {
  padding-top: 69px;
}

.favorite {
  opacity: 0;
  transition: opacity 200ms ease-in;
}

.delivery-fav {
  display: inline-block;
  margin-right: 5px;
  color: gold;
}

.delivery-time-text
 {
  margin-top: -3px;
  margin-left: 4px;
}

.dot-spacing {
  margin-top: -7px;
  margin-left: 4px;
}

.details-location {
  flex: 1;
  color: $color-dark-gray;
  word-break: break-all;
}

.rating {
  background-color: rgba(0, 198, 208, 0.1);
  display: inline-flex;
  color: #00c6d0;
  font-size: 11px;
  font-weight: 500;
  text-transform: uppercase;
  letter-spacing: 0.6px;
  padding: 0 14px;
  align-items: center;
  height: 24px;
  margin: -3px 0 -3px 8px;
}

.icon {
  align-self: center;
  height: 14px;
  width: 14px;
  margin-left: 8px;
  vertical-align: -2px;
  display: inline-flex;
  align-items: center;
  justify-content: center;
}

.overlay-bottom, .overlay-top {
  margin: 24px 24px 0;
  max-width: 164px;
}

.overlay-bottom {
  position: absolute;
  bottom: 24px;
}

.overlay-top {
  position: absolute;
  top: 0;
}

.overlay-h4s {
  font-size: 18px;
  font-weight: 600;
  text-transform: capitalize;
  line-height: normal;
  letter-spacing: -0.2px;
}

.overlay-h4, .overlay-top h4, .overlay-top h5, .overlay-bottom h5
{
  font-family: 'Roboto', sans-serif;
}

.overlay-top h4 {
  text-transform: capitalize
}

.overlay-h4, .overlay-top h4 {
  color: black;
  font-size: 24px;
}

.overlay-top h5 {
  color: black;
  font-size: 16px;
  @media (max-width: 992px) {
    font-size: 14px
  }
  @media (max-width: 767px) {
    font-size: 14px
  }
}

.overlay-bottom h5 {
  color: gray;
  font-size: 13px
}

@media screen and (min-width: 768px) {
  .overlay-bottom, .overlay-top {
    max-width: 190px;
  }
}

@media screen and (max-width: 444px) {
  .overlay-bottom, .overlay-top {
    margin-left: 15px;
    margin-top: 10px
  }

  .overlay-h4, .overlay-top h4 {
    font-size: 16px;
  }

  .overlay-top h5, .overlay-bottom h5 {
    font-size: 10px
  }
}

.img-small-screen {
  display: block;
  @media screen and (min-width: 576px) {
    display: none
  }
}

.img-large-screen {
  display: block;
  @media screen and (max-width: 576px) {
    display: none
  }
}
