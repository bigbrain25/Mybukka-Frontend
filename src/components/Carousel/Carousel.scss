@mixin control {
  outline: none;
  background: white;
  width: 40px;
  height: 40px;
  border-radius: 50%;
  text-align: center;
  margin: auto;
  font-weight: 400;
  border: 0.2px solid #d9dbe0;
  box-shadow: rgba(0, 0, 0, 0.1) 0px 2px 24px 0px;
  position: absolute;
  z-index: 2;
  align-self: center;
  margin: 4px -40px 0;
  cursor: pointer;
  transition: opacity 0.15s ease-in;
}

.pt-29 {
  @media screen and (min-width: 768px) {
    padding-top: 29px;
  }
}

.px-15 {
  padding-left: 15px;
  padding-right: 15px
}

.carousel-divider {
  margin-bottom: 39px;
  width: 100%;
  height: 16px;
  background-color: rgb(231, 230, 230);
  @media screen and (min-width: 768px) {
    display: none;
  }
}

.custom-flex-nowrap {
  flex-flow: row nowrap;
  justify-content: space-between;
  transition: transform 300ms cubic-bezier(0.86, 0, 0.07, 1);
  @media screen and (max-width: 768px) {
    position: relative;
    overflow-x: auto;
    top: 20px;
  }
}

.first-child-pl-0 {
  padding-right: 10px;
  padding-left: 5px;
  :first-child {
    padding-left: 0;
  }
  :last-child {
    padding-right: 0;
  }
  @media (max-width: 768px) {
    padding: 5px;
  }
}

.overflow-hidden {
  position: relative;
  overflow-x: hidden;
  overflow-y: hidden;
}

.control-left {
  @include control;
  transform: rotate(180deg);
  left: 20px;
  @media (max-width: 768px) {
    display: none;
  }
}

.control-right {
  @include control;
  right: 20px;
  @media (max-width: 768px) {
    display: none;
  }
}

.control-left:hover svg,
.control-right:hover svg {
  outline: none;
  opacity: 0.5;
}

.control-left svg,
.control-right svg {
  position: relative;
  top: 15%;
  left: 13%;
}

.carousel-container {
  margin: auto;
  width: 100%;
  display: flex;
  justify-content: center;
}

.carousel {
  padding-bottom: 20px;
}
