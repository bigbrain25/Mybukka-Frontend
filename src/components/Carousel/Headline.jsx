/* eslint-disable react/prop-types */
import React, { useRef, useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import Navlink from 'Components/navlink/Navlink';
import Container from 'Components/container/Container';
import setPromotionToDisplayAction from 'Redux/setPromotionToDisplayAction';

import './Headline.scss';

// NumItemsToView - number of items to view
const NumItemsToView = ({ items, description, id, handleClick }) => (
  <Navlink
    classNames={`headline-link ${description ? 'd-sm-none' : ''}`}
    href={`/place-groups/d/${id}`}
    onClick={handleClick}
    rel="nofollow"
  >
    <span className="d-none pr-3 d-sm-inline-flex">
      View all {items}
    </span>
    <i className="fas fa-chevron-right" />
  </Navlink>
);

const Headline = ({
  title,
  numberOfViews,
  slidesLength,
  activeIndex,
  description,
  placeId,
  setPromotionToDisplay,
  itemSizes,
  useScroll,
  currentTitle,
  setCurrentTitle
}) => {
  const headline = useRef();
  const [state, setState] = useState(0);
  const [result, calc] = useState(0);

  // TODO: cleanup this function
  const updateSlideWidth = (width) => {
    let slideWidth = ((width || state) / (slidesLength - itemSizes)) * activeIndex;
    if (activeIndex === 1) {
      slideWidth = state / (slidesLength - itemSizes);
    } else if (activeIndex > 1) {
      slideWidth = (((width || state) / (slidesLength - itemSizes)) * activeIndex) - 32;
    }
    calc(slideWidth);
  };

  const _inViewport = () => {
    const imageTopPosition = headline.current.getBoundingClientRect().top;

    const buffer = 135;
    if (buffer > imageTopPosition) {
      return true;
    }
    return false;
  };

  const handleScroll = () => {
    if (headline.current && _inViewport() && title !== currentTitle) {
      setCurrentTitle(title);
    }
  };

  useEffect(() => {
    const width = headline ? headline.current.clientWidth : 0;
    setState(width);
    updateSlideWidth(width);

    if (useScroll) {
      document.addEventListener('scroll', handleScroll);
      return () => {
        document.removeEventListener('scroll', handleScroll);
      };
    }
  }, [headline]);

  // TODO: Replace the api call with new API hook
  const handleClick = async () => {
    await setPromotionToDisplay(placeId, title, description);
  };

  const resolveValidId = () => title.replace(/ /g, '-').replace(/'/g, '-').replace(/â‚¦/g, '-');

  // TODO: cleanup, find out if there is need for the effect
  useEffect(() => {
    updateSlideWidth();
  }, [activeIndex]);

  return (
    <Container>
      <div className="headline" id={resolveValidId()} ref={headline}>
        <div
          className="runner h"
          style={{
            transform: `translateX(${result}px)`,
            opacity: 1
          }}
        />
        <div className="d-flex justify-content-between">
          <h2 className="headline-h2">{title}</h2>
          {numberOfViews > 10 &&
            <NumItemsToView
              items={numberOfViews}
              description={description}
              id={placeId}
              handleClick={handleClick}
            />
          }
        </div>
        {description &&
          <p className="d-flex justify-content-between">
            {description}
            <div className="d-none d-sm-inline-flex">
              {numberOfViews > 10 &&
              <NumItemsToView
                items={numberOfViews}
                id={placeId}
                handleClick={handleClick}
              />
              }
            </div>
          </p>
        }
      </div>
    </Container>
  );
};

export default connect(() => ({}), {
  setPromotionToDisplay: setPromotionToDisplayAction,
})(Headline);

Headline.defaultProps = {
  numberOfViews: null,
  slidesLength: 1,
};

Headline.propTypes = {
  numberOfViews: PropTypes.number,
  title: PropTypes.string.isRequired,
  slidesLength: PropTypes.number,
  activeIndex: PropTypes.number.isRequired,
};
