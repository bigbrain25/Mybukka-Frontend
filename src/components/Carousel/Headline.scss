$fontFamily: 'Roboto', sans-serif;

.headline {
  position: relative;
  display: flex;
  justify-content: space-between;
  border-bottom: 1px solid rgba(217, 219, 224, 0.5);
  align-items: flex-end;
  font-family: $fontFamily;
  font-size: 24px;
  letter-spacing: -1.16px;
  font-weight: 600;
  line-height: normal;
  color: rgb(45, 49, 56);
  padding-bottom: 16px;
  text-transform: capitalize;
  margin-bottom: 20px;
  margin-top: 5px;
  @media (max-width: 991.98px) {
    margin-left: 0px !important;
    margin-right: 0px !important;
  }
}

@media screen and (min-width: 1060px) {
  .headline-h2 {
    font-family: $fontFamily;
    font-size: 24px !important;
    letter-spacing: -1.16px;
    font-weight: 500;
    line-height: normal;
  }
}

.runner {
  position: absolute;
  top: 2px;
  transition: left 300ms cubic-bezier(0.86, 0, 0.07, 1);
  width: 32px;
  height: 100%;
  border-bottom: 2px solid rgba(0, 0, 0, 0.9);
  z-index: 200;
}

.headline-h2 {
  font-size: 18px;
  color: #2d3138;
  order: 0;
  margin-bottom: -5px;
}

.headline-link {
  color: #8f95a3;
  font-family: $fontFamily;
  font-size: 14px;
  letter-spacing: 0.14px;
  font-weight: 400;
  align-items: center;
  cursor: pointer;
  flex-shrink: 0;
  flex-grow: 0;
  min-height: 24px;
  padding-top: 5px;
}
