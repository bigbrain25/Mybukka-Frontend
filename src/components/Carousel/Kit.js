/* eslint-disable react/prop-types */
import React, { Fragment } from 'react';

import PropTypes from 'prop-types';

import MarkIcon from 'Icons/Remark';
import Price from 'Components/badge/Price';


const randomNum = tagLenght => Math.floor(Math.random() * tagLenght);

const Rating = ({ tags }) => (
  <span className="rating">{tags ?
    tags[randomNum(tags.length)] : 'POPULAR'}
  </span>
);

const Remark = () => (
  <span>
    <span className="icon">
      <MarkIcon />
    </span>
  </span>
);

const RestauranDetails = ({ name, remark }) => (
  <span className="details-location">
    {name}
    {remark && <Remark />}
  </span>
);

const TopOverlayText = ({ heading, subHeading }) => (
  <div className="overlay-top">
    <h4>{heading}</h4>
    <h5>{subHeading}</h5>
  </div>
);

const BottomOverlayText = ({ heading, subHeading }) => (
  <div className="overlay-bottom">
    <h4 className={
      `${subHeading ? 'overlay-h4' : 'overlay-h4s'}`}
    >{heading}</h4>
    {subHeading && <h5>{subHeading}</h5>}
  </div>
);

const TextOverlay = ({ top, bottom, heading, subHeading, textOverlay }) => (
  textOverlay &&
  <Fragment>
    {top && <TopOverlayText heading={heading} subHeading={subHeading} />}
    {bottom && <BottomOverlayText heading={heading} subHeading={subHeading} />}
  </Fragment>
);

const NormalText = ({
  remark, mealName, deliveryCost, tags,
  textOverlay,
  type,
}) => (
  !textOverlay &&
  <div className="details">
    <h3>
      <RestauranDetails name={mealName} remark={remark} />
      {tags.length > 0 && tags[0] !== null && <Rating tags={tags} />}
    </h3>
    <div className="delivery">
      {deliveryCost ? <Price price={deliveryCost} /> : null}
      {(type !== 'category' && deliveryCost) &&
      <span className="custom-delivery-badge">Delivery</span>
      }
    </div>
  </div>
);

export default {
  NormalText,
  TextOverlay,
};

NormalText.defaultProps = {
  deliveryTime: '',
  rating: '',
  delivery: false,
  deliveryCost: 0,
  mealName: '',
  remark: false,
  tags: []
};

NormalText.propTypes = {
  tags: PropTypes.arrayOf(PropTypes.string),
  remark: PropTypes.oneOfType([PropTypes.string, PropTypes.bool]),
  mealName: PropTypes.string,
  deliveryCost: PropTypes.number,
};

TextOverlay.propTypes = {
  top: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]).isRequired,
  bottom: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]).isRequired,
  heading: PropTypes.string.isRequired,
  subHeading: PropTypes.string.isRequired
};

TopOverlayText.propTypes = {
  heading: PropTypes.string.isRequired,
  subHeading: PropTypes.string.isRequired
};

BottomOverlayText.defaultProps = {
  subHeading: ''
};

BottomOverlayText.propTypes = {
  heading: PropTypes.string.isRequired,
  subHeading: PropTypes.string,
};

