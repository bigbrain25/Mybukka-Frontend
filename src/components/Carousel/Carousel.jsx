import React, { useState } from 'react';
import ChevronRight from 'Icons/ChevronRight';
import PropTypes, { any } from 'prop-types';
import FoodCard from './BukkaCard';
import Headline from './Headline';
import './Carousel.scss';

const ControlLeft = ({ handleClick }) => (
  <div
    aria-pressed="false"
    tabIndex="0"
    role="button"
    onClick={handleClick}
    className="control-left"
  >
    <ChevronRight />
  </div>
);

const ControlRight = ({ handleClick }) => (
  <div
    aria-pressed="false"
    tabIndex="0"
    role="button"
    onClick={handleClick}
    className="control-right"
  >
    <ChevronRight />
  </div>
);

const Carousel = ({ classNames, noOfImagesShown, title, NumberOfViews, slideItems }) => {
  const [activeIndex, setActiveIndex] = useState(0);

  // max-width = 100% divide by no of images expected to show at the same time
  const maxWidth = 100;
  const slidesLenght = slideItems.length;
  const translate = activeIndex >= 1 ? maxWidth / noOfImagesShown : 0;

  const goToPrevSlide = (e) => {
    let index = activeIndex;
    e.preventDefault();

    index -= 1;
    setActiveIndex(index);
  };

  const goToNextSlide = (e) => {
    let index = activeIndex;
    e.preventDefault();

    index += 1;
    setActiveIndex(index);
  };

  return (
    <div className="mt-4 mb-4">
      <Headline
        activeIndex={activeIndex}
        title={title}
        slidesLenght={slidesLenght}
        NumberOfViews={NumberOfViews}
      />
      <div className="carousel-container">
        {activeIndex >= 1 && <ControlLeft handleClick={goToPrevSlide} />}
        <div className="overflow-hidden">
          <div
            style={
              { transform: `translateX(${activeIndex * -translate}%)` }}
            className="row pb-4 flex-nowrap"
          >
            { slideItems.map(bukka =>
              (<FoodCard
                image={bukka.image}
                deliveryCost={bukka.deliveryCost}
                deliveryTime={bukka.deliveryTime}
                rating={bukka.rating}
                classNames={classNames}
                imageCount={noOfImagesShown === '2' ? '' : noOfImagesShown}
              />))}
          </div>
        </div>
        { activeIndex + Number(noOfImagesShown) !== slidesLenght
        && <ControlRight handleClick={goToNextSlide} />}
      </div>
    </div>
  );
};

export default Carousel;

ControlLeft.propTypes = {
  handleClick: PropTypes.func.isRequired,
};

ControlRight.propTypes = {
  handleClick: PropTypes.func.isRequired,
};

Carousel.propTypes = {
  classNames: PropTypes.string.isRequired,
  slideItems: PropTypes.arrayOf(any).isRequired,
  title: PropTypes.string.isRequired,
  NumberOfViews: PropTypes.number.isRequired,
  noOfImagesShown: PropTypes.string.isRequired
};
