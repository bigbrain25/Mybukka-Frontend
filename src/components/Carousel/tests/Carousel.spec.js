import React from 'react';

import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

import bukkaData from 'Features/feed/common/bukkaData.json';
import Carousel from '../Carousel';

const initialState = {
  navbarAuthReducer: { type: 'Sign In', },
  deliveryModeReducer: { mode: 'delivery', },
  locationsPredictionReducer: { predictions: [3.7474, 3.4848] },
  authenticationReducer: {
    status: {
      authenticated: false,
      error: false
    }
  },
  cuisineReducer: {
    cuisineItems: [{
      _id: '5ddc5dfb9a29af135cd21ce7',
      name: 'american',
      imageUrl: 'https://res.cloudinary.com/mybukka/image/upload/v1574723013/American_e1n3aj.png'
    }],
    errorMessage: '',
    currentPage: 1,
    cuisineToDisplay: { name: 'cuisine name' },
  },
  bukkasReducer: {
    fetchedBukkas: {
      nearbyBukkas: [],
      message: ''
    },
    status: {
      fetchedBukkas: false,
      error: true,
    }
  },
  selectedLocationReducer: { coordinates: [3.7474, 3.4848] }
};

const store = mockStore(initialState);

describe('Carousel component', () => {
  const props = {
    textOverlay: false,
    classNames: 'some-classnames',
    noOfImagesShown: 3,
    title: 'A carousel',
    slideItems: [
      {
        deliveryTime: '30-50 min',
        author: 'Bane',
        deliveryCost: 300,
        imageUrl:
          'https://res.cloudinary.com/deqt3envc/image/upload/v1549300439/banner-img-3.jpg',
        rating: 'popular',
        position: { top: true },
        heading: 'it\'s heading',
        subHeading: 'sub heading here',
      },
      {
        deliveryTime: '1 hour',
        author: "Ra's Al Ghul",
        deliveryCost: 400,
        imageUrl:
          'https://res.cloudinary.com/deqt3envc/image/upload/v1549300440/banner-img-2.jpg',
        position: { bottom: true },
        heading: 'it\'s heading',
        subHeading: 'sub heading here',
      }
    ],
    imageHeight: 'img-height',
    controlClassNames: 'control-class'
  };

  const { container, rerender } = render(
    <Provider store={store}>
      <MemoryRouter>
        <Carousel {...props} />
      </MemoryRouter>
    </Provider>
  );

  it('renders properly', () => {
    expect(container).toMatchSnapshot();
  });

  it('renders properly', () => {
    const nearbyBukkas = bukkaData;
    const newState = {
      ...initialState,
      bukkasReducer: {
        ...initialState.bukkasReducer,
        fetchedBukkas: {
          ...initialState.bukkasReducer.fetchedBukkas,
          nearbyBukkas,
          message: ''
        },
        status: {
          ...initialState.bukkasReducer.status,
          fetchedBukkas: true,
          error: false
        }
      }
    };
    const newProps = {
      ...props,
      fetchedBukkas: { nearbyBukkas },
    };
    const newStore = mockStore(newState);
    rerender(
      <Provider store={newStore}>
        <MemoryRouter>
          <Carousel {...newProps} />
        </MemoryRouter>
      </Provider>
    );
    expect(container).toMatchSnapshot();
  });
});
