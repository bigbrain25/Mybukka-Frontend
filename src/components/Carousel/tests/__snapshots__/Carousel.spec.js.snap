// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Carousel component renders properly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="container "
      >
        <div
          class="headline"
        >
          <div
            class="runner"
            style="transform: translateX(0% - 32px);"
          />
          <h2
            class="headline-h2"
          >
            A carousel
          </h2>
        </div>
      </div>
      <div
        class="container "
      >
        <div
          class="carousel"
        >
          <div
            class="carousel-container"
          >
            <div
              class="overflow-hidden"
            >
              <div
                class="row custom-flex-nowrap"
                style="transform: translateX(0%);"
              >
                <div
                  class="card-container some-classnames"
                >
                  <div
                    class="card-wrap"
                  >
                    
                    <div
                      class="mt-4 bukka-card"
                      data-target=""
                      data-toggle=""
                    >
                      <img
                        alt="alt_image"
                        class="img-small-screen bukka-img img-height"
                        src="https://res.cloudinary.com/deqt3envc/image/upload/c_scale,w_320,h_320/v1549300439/banner-img-3.jpg"
                      />
                      <img
                        alt="alt_image"
                        class="img-large-screen bukka-img img-height"
                        src="https://res.cloudinary.com/deqt3envc/image/upload/c_scale,w_640,h_640/v1549300439/banner-img-3.jpg"
                      />
                      <div
                        class="details"
                      >
                        <h3>
                          <span
                            class="details-location"
                          >
                            
                          </span>
                        </h3>
                        <div
                          class="delivery"
                        >
                          0
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
                <div
                  class="card-container some-classnames"
                >
                  <div
                    class="card-wrap"
                  >
                    
                    <div
                      class="mt-4 bukka-card"
                      data-target=""
                      data-toggle=""
                    >
                      <img
                        alt="alt_image"
                        class="img-small-screen bukka-img img-height"
                        src="https://res.cloudinary.com/deqt3envc/image/upload/c_scale,w_320,h_320/v1549300440/banner-img-2.jpg"
                      />
                      <img
                        alt="alt_image"
                        class="img-large-screen bukka-img img-height"
                        src="https://res.cloudinary.com/deqt3envc/image/upload/c_scale,w_640,h_640/v1549300440/banner-img-2.jpg"
                      />
                      <div
                        class="details"
                      >
                        <h3>
                          <span
                            class="details-location"
                          >
                            
                          </span>
                        </h3>
                        <div
                          class="delivery"
                        >
                          0
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="container "
    >
      <div
        class="headline"
      >
        <div
          class="runner"
          style="transform: translateX(0% - 32px);"
        />
        <h2
          class="headline-h2"
        >
          A carousel
        </h2>
      </div>
    </div>
    <div
      class="container "
    >
      <div
        class="carousel"
      >
        <div
          class="carousel-container"
        >
          <div
            class="overflow-hidden"
          >
            <div
              class="row custom-flex-nowrap"
              style="transform: translateX(0%);"
            >
              <div
                class="card-container some-classnames"
              >
                <div
                  class="card-wrap"
                >
                  
                  <div
                    class="mt-4 bukka-card"
                    data-target=""
                    data-toggle=""
                  >
                    <img
                      alt="alt_image"
                      class="img-small-screen bukka-img img-height"
                      src="https://res.cloudinary.com/deqt3envc/image/upload/c_scale,w_320,h_320/v1549300439/banner-img-3.jpg"
                    />
                    <img
                      alt="alt_image"
                      class="img-large-screen bukka-img img-height"
                      src="https://res.cloudinary.com/deqt3envc/image/upload/c_scale,w_640,h_640/v1549300439/banner-img-3.jpg"
                    />
                    <div
                      class="details"
                    >
                      <h3>
                        <span
                          class="details-location"
                        >
                          
                        </span>
                      </h3>
                      <div
                        class="delivery"
                      >
                        0
                      </div>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="card-container some-classnames"
              >
                <div
                  class="card-wrap"
                >
                  
                  <div
                    class="mt-4 bukka-card"
                    data-target=""
                    data-toggle=""
                  >
                    <img
                      alt="alt_image"
                      class="img-small-screen bukka-img img-height"
                      src="https://res.cloudinary.com/deqt3envc/image/upload/c_scale,w_320,h_320/v1549300440/banner-img-2.jpg"
                    />
                    <img
                      alt="alt_image"
                      class="img-large-screen bukka-img img-height"
                      src="https://res.cloudinary.com/deqt3envc/image/upload/c_scale,w_640,h_640/v1549300440/banner-img-2.jpg"
                    />
                    <div
                      class="details"
                    >
                      <h3>
                        <span
                          class="details-location"
                        >
                          
                        </span>
                      </h3>
                      <div
                        class="delivery"
                      >
                        0
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
